### PR TITLE
Rename field name constants in AbstractBuilderTestCase

### PR DESCRIPTION
**Pull request by cbuescher**
Some field name constants were not updaten when we moved from "string" to "text"
and "keyword" fields. Renaming them makes it easier and faster to know which
field type is used in test subclassing this base test case.

### CODE REVIEW

File: test/framework/src/main/java/org/elasticsearch/test/AbstractBuilderTestCase.java [hunk 1]
```diff
@@ -376,9 +377,9 @@ public void onRemoval(ShardId shardId, Accountable accountable) {
 
             if (registerType) {
                 mapperService.merge("_doc", new CompressedXContent(Strings.toString(PutMappingRequest.simpleMapping(
-                    STRING_FIELD_NAME, "type=text",
-                    STRING_FIELD_NAME_2, "type=keyword",
-                    STRING_ALIAS_FIELD_NAME, "type=alias,path=" + STRING_FIELD_NAME,
+                    TEXT_FIELD_NAME, "type=text",
+                    KEYWORD_FIELD_NAME, "type=keyword",
+                    TEXT_ALIAS_FIELD_NAME, "type=alias,path=" + TEXT_FIELD_NAME,
```
Comment (cbuescher) [inline test/framework/src/main/java/org/elasticsearch/test/AbstractBuilderTestCase.java:382]: This is the most relevant change, the rest should be just renames according to this part.

File: test/framework/src/main/java/org/elasticsearch/test/AbstractBuilderTestCase.java [hunk 2]
```diff
@@ -104,9 +105,9 @@
 
 public abstract class AbstractBuilderTestCase extends ESTestCase {
 
-    public static final String STRING_FIELD_NAME = "mapped_string";
-    public static final String STRING_ALIAS_FIELD_NAME = "mapped_string_alias";
-    protected static final String STRING_FIELD_NAME_2 = "mapped_string_2";
+    public static final String TEXT_FIELD_NAME = "mapped_string";
+    public static final String TEXT_ALIAS_FIELD_NAME = "mapped_string_alias";
+    protected static final String KEYWORD_FIELD_NAME = "mapped_string_2";
```
Comment (cbuescher) [inline test/framework/src/main/java/org/elasticsearch/test/AbstractBuilderTestCase.java:110]: Renaming here, according to the field types used below

**Comment by cbuescher** [comment]
@elasticmachine update branch

**Review by javanna — APPROVED** [review]
LGTM

**Pull request merged**